tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -409,15 +372,6 @@ StorageManagerCanonical::array_load_fragments(
     Array* array, const std::vector<TimestampedURI>& fragments_to_load) {
   auto timer_se = stats_->start_timer("array_load_fragments");
 
-  // Check if the array is open
-  auto it = open_arrays_.find(array);
```

Comment:
I don't think we can guarantee that the `array` is in a usable state here (at least not yet), and none of the calls in the following block actually check whether the array is open (so eg `array_schema_latest_ptr()` could be null).

So we need to either keep an `array->is_open()` check here, or add check-and-throw in each of the calls.

cc @bekadavis9

Reply:
Unless I'm mistaken, this function is only ever called from one location [1] in `tiledb/sm/array/array.cc` which means that if we have a broken pre-condition its something that should be covered there. Granted I can go ahead and add those checks as well. Also, this is just PR is mostly just preparatory for moving these array functions to array.cc so that we can remove StorageManager dependencies from the Array class completely.

[1] https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/array/array.cc#L220